File: S01_AuthUI/app/src/main/java/com/example/s01_authui/RegisterActivity.java
package com.example.s01_authui;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatButton;

public class RegisterActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.register_layout);

        AppCompatButton pageLoginBtn = findViewById(R.id.page_login_btn);
        pageLoginBtn.setOnClickListener(v -> {
            Intent intent = new Intent(RegisterActivity.this, MainActivity.class);
            startActivity(intent);
        });

        ImageButton registerBtn = findViewById(R.id.register_post_btn);

        registerBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                EditText usernameEt = findViewById(R.id.username_register_et);
                EditText emailEt = findViewById(R.id.email_register_et);
                EditText passwordEt = findViewById(R.id.password_register_et);

                String username = usernameEt.getText().toString();
                String email = emailEt.getText().toString();
                String password = passwordEt.getText().toString();

                SharedPreferences prefs = getSharedPreferences("user_prefs", MODE_PRIVATE);
                SharedPreferences.Editor editor = prefs.edit();
                editor.putString("username", username);
                editor.putString("email", email);
                editor.putString("password", password);
                editor.apply();

                Toast.makeText(RegisterActivity.this, "Registro exitoso", Toast.LENGTH_SHORT).show();

            }
        });


    }
}

File: S02_UNSFormRegister/app/src/main/java/com/example/s02_unsformregister/RegisterActivity.kt
package com.example.s02_unsformregister

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.ImageButton
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class RegisterActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.register_activity)

        val btnHome = findViewById<ImageButton>(R.id.back_main)
        btnHome.setOnClickListener{
            finish()
        }

        val etCodigo = findViewById<EditText>(R.id.etCodigo)
        val etNombres = findViewById<EditText>(R.id.etNombres)
        val etApellidos = findViewById<EditText>(R.id.etApellidos)
        val etCorreo = findViewById<EditText>(R.id.etCorreo)
        val btnRegistrar = findViewById<Button>(R.id.btnRegistrar)

        btnRegistrar.setOnClickListener {
            val codigo = etCodigo.text.toString().trim()
            val nombres = etNombres.text.toString().trim()
            val apellidos = etApellidos.text.toString().trim()
            val correo = etCorreo.text.toString().trim()

            if (codigo.isEmpty() || nombres.isEmpty() || apellidos.isEmpty() || correo.isEmpty()) {
                Toast.makeText(this, "Por favor completa todos los campos", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            val estudiante = Student(codigo, apellidos, nombres, correo)
            StudentStorage.students.add(estudiante)

            Toast.makeText(this, "Estudiante registrado con éxito", Toast.LENGTH_SHORT).show()

            etCodigo.text.clear()
            etNombres.text.clear()
            etApellidos.text.clear()
            etCorreo.text.clear()
        }
    }

    override fun finish() {
        super.finish()
        overridePendingTransition(R.anim.slide_in_down, R.anim.slide_out_down)
    }
}